functions.py

```diff
@@ -506,15 +555,27 @@ def sub(x, y):
 
 
 def tan(x):
-    return math.tan(_scale_angle(x))
+    if(_scale_angle(x) % math.pi == 0):
+        return 0
+    x = math.tan(_scale_angle(x))
+    n = significant(x)
+    if(n > 15):
```

Comment:
How did you choose 15 ?

Reply:
@pro-panda that is because, on executing tan(90), the value that is displayed is of the order of 10^15. Hence I chose the number as 15.